src/main.c

```diff
@@ -132,8 +132,8 @@ static bool parse_args(int argc, char **args)
                                strlen(prog_basename) + 5 + 1);
         assert(prof_out_file);
 
-        sprintf(prof_out_file, "%s/%s%s.prof", cwd_path, rel_path,
-                prog_basename);
+        snprintf(prof_out_file, sizeof(prof_out_file), "%s/%s%s.prof", cwd_path,
```

Comment:
I believe that sizeof(prof_out_file) is incorrect. IIUC, it will always return the size of the pointer (most likely 4 or 8).

Reply:
Corrected